### PR TITLE
Refactor: Introduce complexConjugate Message, Deprecate conjugated

### PR DESCRIPTION
**Pull request by hemalvarambhia**
Issue no. #192.

### CODE REVIEW

File: src/Math-Complex/PMComplex.class.st [hunk 1]
```diff
@@ -544,6 +557,12 @@ PMComplex >> isNumber [
 	^ true
 ]
 
+{ #category : #testing }
+PMComplex >> isReal [
+
+	^ imaginary = 0 and: [ real >= 0 ]
+]
```
Comment (SergeStinckwich) [inline src/Math-Complex/PMComplex.class.st:564]: 0 is not real?

File: src/Math-Complex/PMComplex.class.st [hunk 2]
```diff
@@ -694,14 +713,14 @@ PMComplex >> sinh [
 
 { #category : #'mathematical functions' }
 PMComplex >> sqrt [
+
 	"Return the square root of the receiver with a positive imaginary part."
 
 	| u v |
-	(imaginary = 0 and: [real >= 0])
-		ifTrue:	[^self class real: real sqrt imaginary: 0].
+	self isReal ifTrue: [ ^ self class real: real sqrt imaginary: 0 ].
```
Comment (SergeStinckwich) [inline src/Math-Complex/PMComplex.class.st:720]: I think you extract ``(imaginary = 0 and: [real >= 0])`` a bit too fast, this is not replaced by isReal but by isReal and positive.

Comment (hemalvarambhia) [reply src/Math-Complex/PMComplex.class.st:720]: How about `z isZero or: z isReal` ?

Comment (hemalvarambhia) [reply src/Math-Complex/PMComplex.class.st:720]: How about `z isZero or: z isReal`?

File: src/Math-Complex/PMComplex.class.st (hunk 1 above)
Comment (hemalvarambhia) [reply src/Math-Complex/PMComplex.class.st:564]: Very true. Thus we have a bug.

File: src/Math-Complex/PMComplex.class.st (hunk 2 above)
Comment (SergeStinckwich) [reply src/Math-Complex/PMComplex.class.st:720]: I don't know if this is necessary. Maybe keep it like it was before.

Comment (hemalvarambhia) [reply src/Math-Complex/PMComplex.class.st:720]: Sure. Perhaps we need more tests before refactoring.

**Review by SergeStinckwich — APPROVED** [review]

File: src/Math-Complex/PMComplex.class.st (hunk 2 above)
Comment (hemalvarambhia) [reply src/Math-Complex/PMComplex.class.st]: I have inlined the method for us.

File: src/Math-Complex/PMComplex.class.st [hunk 3]
```diff
@@ -694,14 +707,15 @@ PMComplex >> sinh [
...
 
 	| u v |
-	(imaginary = 0 and: [real >= 0])
-		ifTrue:	[^self class real: real sqrt imaginary: 0].
+	(imaginary = 0 and: [ real >= 0 ]) ifTrue: [ 
+		^ self class real: real sqrt imaginary: 0 ].
 	v := (self abs - real / 2) sqrt.
 	u := imaginary / 2 / v.
-	^self class real: u imaginary: v
+	^ self class real: u imaginary: v
```
Comment (hemalvarambhia) [inline src/Math-Complex/PMComplex.class.st:718]: These may be formatting improvements by Pharo when I inlined the `isReal` method

**Comment by hemalvarambhia** [comment]
@SergeStinckwich I think we could add more tests for the square root message. I can do that as my next PR. Rather than tests in one method, I would rather write separate test methods with good names.

**Pull request merged**